includes/payment-methods/class-upe-payment-gateway.php

```diff
@@ -1074,4 +1110,57 @@ public function log_payment_error_ajax() {
 			);
 		}
 	}
+
+	/**
+	 * Creates a cookie to store the id and client secret of payment intent needed to mount the UPE element in frontend.
+	 *
+	 * An intent is tied to the cart's current state. New or removed items will likely impact the cart value,
+	 * necessitating the creation of a new intent with the revised cart value.
+	 * So a new cookie is expected to be created whenever the cart is modified.
+	 *
+	 * @param string $intent_id     The payment intent id.
+	 * @param string $client_secret The payment intent client secret.
+	 */
+	private function create_upe_payment_intent_cookie( string $intent_id = '', string $client_secret = '' ) {
+		$cart_hash = 'undefined';
+
+		if ( isset( $_COOKIE[ self::COOKIE_CART_HASH ] ) ) {
+			$cart_hash = sanitize_text_field( wp_unslash( $_COOKIE[ self::COOKIE_CART_HASH ] ) );
+		}
+
+		$cookie_val = $cart_hash . '-' . $intent_id . '-' . $client_secret;
+
+		wc_setcookie( self::COOKIE_UPE_PAYMENT_INTENT, $cookie_val, time() + MINUTE_IN_SECONDS * self::COOKIE_UPE_EXPIRES_IN_MINS );
+	}
+
+	/**
+	 * Creates a cookie to store the id and client secret of setup intent needed to mount the UPE element in frontend.
+	 *
+	 * @param string $intent_id     The setup intent id.
+	 * @param string $client_secret The setup intent client secret.
+	 */
+	private function create_upe_setup_intent_cookie( string $intent_id = '', string $client_secret = '' ) {
+		$cookie_val = $intent_id . '-' . $client_secret;
+		wc_setcookie( self::COOKIE_UPE_SETUP_INTENT, $cookie_val, time() + MINUTE_IN_SECONDS * self::COOKIE_UPE_EXPIRES_IN_MINS );
+	}
+
+	/**
+	 * Removes the payment intent cookie created for UPE.
+	 */
+	public static function remove_upe_payment_intent_cookie() {
+		if ( isset( $_COOKIE[ self::COOKIE_UPE_PAYMENT_INTENT ] ) ) {
+			unset( $_COOKIE[ self::COOKIE_UPE_PAYMENT_INTENT ] );
+			setcookie( self::COOKIE_UPE_PAYMENT_INTENT, '', time() - HOUR_IN_SECONDS, '/' );
```

Comment:
No reason. I guess I incorrectly assumed that `setcookie` would work better for removing cookies since it accepted the path param which I read at a few places was more effective in removing cookies.

Fixed in a799644